### ISSUE TITLE
Cannot deploy stages using different profiles in v1.0.0-beta.2

### ISSUE THREAD

**Issue by rowanu** [serverless/serverless#1940]
**Note: This was changed from a feature request to a bug**

Serverless Framework Version: v1.0.0-beta.2

Operating System: OSX

Expected Behavior: Setting the `profile` variable (in `serverless.env.yml`) for a stage makes `sls` commands use it when calling the AWS API.

Actual Behavior: The variable in file is ignored, and only the profile set by environment variable `AWS_PROFILE` is used, regardless of the stage being targeted.

To reproduce:

1. Set a `profile` value (defined in your `~/.aws/credentials` file) in your stage's variables.
1. Keep your `AWS_PROFILE` environment variable unset.
1. Run an `sls` command against the stage with the profile defined in the config file. It will not recognise the config file-based value.

**Comment by rowanu** [comment]
I've worked around this by combining AWS profiles (environment variables) and stages per-command, but it's clunky and manual. I'd really just like to be able to run `sls deploy` and have it take care of things...

**Comment by rowanu** [comment]
There's still support for *reading* the `AWS_[STAGE]_PROFILE` variables, but they still aren't recognised by the SDK.

**Comment by rowanu** [comment]
To the maintainers: sorry for all the edits, I did the first version of this ticket in a rush.

I had a PR to fix this bug but it was rejected due to upcoming changes. I'd be happy to update it once #1834 is merged.

**Comment by alexandrusavin** [comment]
@rowanu I guess #1834 has been merged. Could you update your PR?
Thx!

**Comment by rowanu** [comment]
I'm just waiting on feedback how to change it, see #1945.

**Comment by rowanu** [comment]
Closing in favour of #2015.
